hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+#include "hamilton/hybrid_planner.h"
+
+//TODO namespace
+//TODO bools aren't returning anything!
+namespace hamilton
+{
+	
+	HybridPlanner::~HybridPlanner()
+	{
+		moveit_group_ptr_.reset();
+		kinematic_state_.reset();
+		//TODO Descartes
+	}
+
+	void HybridPlanner::init()
+	{
+		// Instantiating a robot model
+		descartes_robot_model_ptr_.reset(new descartes_moveit::MoveitStateAdapter);
+		// TOCHECK as compared to: 
+		//descartes_core::RobotModelPtr model(new descartes_moveit::MoveitStateAdapter);
+		//TOCHECK or as in Godel
+		// init descartes with kinematic_state_ ? 
+		//   descartes_core::RobotModelPtr createRobotModel(const moveit::core::RobotStatePtr robot_state,
+                                                 // const std::string& group_name,
+                                                 // const std::string& tool_frame,
+                                                 // const std::string& world_frame,
+                                                 // uint8_t iterations)
+		if(descartes_robot_model_ptr_->initialize(ROBOT_DESCRIPTION_PARAM,
+		                              config_.group_name,
+		                              config_.world_frame,
+		                              config_.tip_link))
+		{
+		ROS_INFO_STREAM("Descartes Robot Model initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Robot Model");
+		exit(-1);
+		}
+
+		// TODO add planner type as config parameter?
+		bool succeeded_dense = dense_planner_.initialize(descartes_robot_model_ptr_);
+		if(succeeded_dense)
+		{
+		ROS_INFO_STREAM("Descartes dense planner initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Descartes dense Planner");
+		exit(-1);
+		}  
+
+		bool succeeded_sparse = sparse_planner_.initialize(descartes_robot_model_ptr_);
+		if(succeeded_sparse)
+		{
+		ROS_INFO_STREAM("Descartes sparse planner initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Descartes sparse Planner");
+		exit(-1);
+		}
+
+		ROS_INFO_STREAM("Descartes members initialized");
+
+		// creates publisher for trajectory visualization
+		marker_publisher_  = nh_.advertise<visualization_msgs::MarkerArray>(VISUALIZE_TRAJECTORY_TOPIC,1,true);
+
+		//Initializes MoveIt!
+		moveit_group_ptr_ = MoveGroupPtr(new moveit_group_interface::MoveGroup(config_.group_name));
```

Comment:
I am using it in `HybridPlanner::appendFreeSpaceSegment(std::vector<geometry_msgs::Pose> waypoints)` here https://github.com/ros-industrial-consortium/hamilton/pull/1/files#r33507832#L 317-320